sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/implementation/AzureEnvironment.java

```diff
@@ -0,0 +1,352 @@
+            put("sqlServerHostnameSuffix", ".database.cloudapi.de");
+            put("galleryEndpointUrl", "https://gallery.cloudapi.de/");
+            put("activeDirectoryEndpointUrl", "https://login.microsoftonline.de/");
+            put("activeDirectoryResourceId", "https://management.core.cloudapi.de/");
+            put("activeDirectoryGraphResourceId", "https://graph.cloudapi.de/");
+            put("microsoftGraphResourceId", "https://graph.microsoft.de/");
+            // TODO: add resource id for the germany cloud for datalake once it is defined.
+            put("dataLakeEndpointResourceId", "N/A");
+            put("activeDirectoryGraphApiVersion", "2013-04-05");
+            put("storageEndpointSuffix", ".core.cloudapi.de");
+            put("keyVaultDnsSuffix", ".vault.microsoftazure.de");
+            // TODO: add dns suffixes for the germany cloud for datalake store and datalake analytics once they are
+            //  defined.
+            put("azureDataLakeStoreFileSystemEndpointSuffix", "N/A");
+            put("azureDataLakeAnalyticsCatalogAndJobEndpointSuffix", "N/A");
+            put("azureLogAnalyticsResourceId", "N/A");
+            put("azureApplicationInsightsResourceId", "N/A");
+        }
+    });
+
+    /**
+     * @return the entirety of the endpoints associated with the current environment.
+     */
+    public Map<String, String> getEndpoints() {
+        return endpoints;
+    }
+
+    /**
+     * @return the list of known environments to Azure SDK.
+     */
+    public static List<AzureEnvironment> knownEnvironments() {
+        return Arrays.asList(AZURE, AZURE_CHINA, AZURE_GERMANY, AZURE_US_GOVERNMENT);
+    }
+
+    /**
+     * @return the management portal URL.
+     */
+    public String getPortal() {
+        return endpoints.get("portalUrl");
+    }
+
+    /**
+     * @return the publish settings file URL.
+     */
+    public String getPublishingProfile() {
+        return endpoints.get("publishingProfileUrl");
+    }
+
+    /**
+     * @return the management service endpoint.
+     */
+    public String getManagementEndpoint() {
+        return endpoints.get("managementEndpointUrl");
+    }
+
+    /**
+     * @return the resource management endpoint.
+     */
+    public String getResourceManagerEndpoint() {
+        return endpoints.get("resourceManagerEndpointUrl");
+    }
+
+    /**
+     * @return the sql server management endpoint for mobile commands.
+     */
+    public String getSqlManagementEndpoint() {
+        return endpoints.get("sqlManagementEndpointUrl");
+    }
+
+    /**
+     * @return the dns suffix for sql servers.
+     */
+    public String getSqlServerHostnameSuffix() {
+        return endpoints.get("sqlServerHostnameSuffix");
+    }
+
+    /**
+     * @return the Active Directory login endpoint.
+     */
+    public String getActiveDirectoryEndpoint() {
+        return endpoints.get("activeDirectoryEndpointUrl").replaceAll("/$", "") + "/";
```

Comment:
If this is the AD endpoint we should insert it into the map with this replacement instead of on retrieval

Reply:
removed this API.